src/app/Fake.DotNet.NuGet/NuGet.fs

```diff
@@ -315,40 +327,113 @@ let private pack parameters nuspecFile =
+      NoServiceEndpoint: bool
+      /// Specifies the server URL. This option is required unless DefaultPushSource config value is set in the NuGet config file.
+      Source: string option
+      /// The API key for the symbol server.
+      SymbolApiKey: string option
+      /// Specifies the symbol server URL.
+      SymbolSource: string option
+      /// Specifies the timeout for pushing to a server.
+      Timeout: TimeSpan option
+      /// Number of times to retry pushing the package
+      PushTrials: int}
+
+    static member Create() =
+        { DisableBuffering = false
+          ApiKey = None
+          NoSymbols = false
+          NoServiceEndpoint = false
+          Source = None
+          SymbolApiKey = None
+          SymbolSource = None
+          Timeout = None
+          PushTrials = 5 }
+
+type NuGetParams with
+    member internal x.NuGetPushOptions = 
+        let normalize str = if String.isNullOrEmpty str then None else Some str
+    
+        { DisableBuffering = false
+          ApiKey = normalize x.AccessKey
+          NoSymbols = false
+          NoServiceEndpoint = false
+          Source = normalize x.PublishUrl
+          SymbolApiKey = normalize x.SymbolAccessKey
+          SymbolSource = normalize x.SymbolPublishUrl
+          Timeout = Some x.TimeOut
+          PushTrials = x.PublishTrials }
+
+    member internal x.ToolOptions = ToolOptions.Create x.ToolPath "push" x.WorkingDir true
+    member internal x.Nupkg = (x.OutputPath @@ packageFileName x |> Path.getFullName)
+
+let private toPushCliArgs param =
+    let ifTrue x b =
+        if b then Some x
+        else None
+
+    [
+        param.ApiKey
+        param.DisableBuffering |> ifTrue "-DisableBuffering"
+        param.NoSymbols |> ifTrue "-NoSymbols"
+        param.NoServiceEndpoint |> ifTrue "-NoServiceEndpoint"
+        param.Source |> Option.map (sprintf "-Source %s")
+        param.SymbolApiKey |> Option.map (sprintf "-SymbolApiKey %s")
+        param.SymbolSource |> Option.map (sprintf "-SymbolSource %s")
+        param.Timeout |> Option.map string |> Option.map (sprintf "-Timeout %s")
+    ]
+    |> List.choose id
+    |> String.concat " "
+
+module Private =
+    /// For internal use
+    let rec push (options : ToolOptions) (parameters : NuGetPushParams) (toCliArgs : NuGetPushParams -> string) nupkg =
+        parameters.ApiKey |> Option.iter (fun key -> TraceSecrets.register key "<NuGetKey>")
+        parameters.SymbolApiKey |> Option.iter (fun key -> TraceSecrets.register key "<NuGetSymbolKey>")
+    
+        let args = sprintf "%s \"%s\" %s" options.Command nupkg (toCliArgs parameters)
+
+        sprintf "%s %s in WorkingDir: %s Trials left: %d" options.ToolPath args (Path.getFullName options.WorkingDir) parameters.PushTrials
+        |> TraceSecrets.guardMessage
+        |> Trace.trace
+    
+        try
+            let result =
+                let tracing = Process.shouldEnableProcessTracing()
```

Comment:
If we really want to disable tracing we should use the corresponding `CreateProcess` functionality. We should remove this class to `Process.shouldEnableProcessTracing` to reduce side effects.

Reply:
Right, I just copied some other code in the same module. I can just remove it I guess.